### PR TITLE
feat(PoExtractor): add single file output

### PR DESCRIPTION
**Pull request by VincenzoCarlino**
Implementation of: #93

### CODE REVIEW

File: src/OrchardCoreContrib.PoExtractor/Program.cs [hunk 1]
```diff
@@ -195,6 +224,7 @@ private static void ShowHelp()
         Console.WriteLine("  -t, --template <Razor|Liquid>          Specifies the template engine to extract the translatable strings from.");
         Console.WriteLine("                                         Default: Razor & Liquid templates");
         Console.WriteLine("  -i, --ignore project1,project2         Ignores extracting PO filed from a given project(s).");
-        Console.WriteLine("  --localizer localizer1,localizer2      Specifies the name of the localizer(s) that will be used suring the extraction process.");
+        Console.WriteLine("  --localizer localizer1,localizer2      Specifies the name of the localizer(s) that will be used during the extraction process.");
+        Console.WriteLine("  -s, --single messages.pot              Get single output file.");
```
Comment (hishamco) [inline src/OrchardCoreContrib.PoExtractor/Program.cs:228]:
```suggestion
        Console.WriteLine("  -s, --single messages.pot              Specifices the single output file.");
```

**Review by hishamco — CHANGES_REQUESTED** [review]
It would be nice to have few unit tests, there're many corner cases that we need to test, to ensure nothing is broken

**Comment by VincenzoCarlino** [comment]
> It would be nice to have few unit tests, there're many corner cases that we need to test, to ensure nothing is broken

Look at this [commit](https://github.com/OrchardCoreContrib/OrchardCoreContrib.PoExtractor/pull/94/commits/479751a4634e4b962a9cd7af95d6c6bdefaaefd4)

It's a preview of some kinds of test we can do.
All code is inside the main static function, so we can test it as I've done it in this commit, or we try to extract the code from main in something more testable.
Let me know if you like it.

**Comment by hishamco** [comment]
Thanks, @VincenzoCarlino, we might need to test more cases such as if there's a PO file with the same specified file

**Comment by VincenzoCarlino** [comment]
> Thanks, @VincenzoCarlino, we might need to test more cases such as if there's a PO file with the same specified file

Yes @hishamco , this is just an example of the structure of tests. Tomorrow I'll add more tests.

**Comment by hishamco** [comment]
Please fix the build

**Comment by VincenzoCarlino** [comment]
Hi @hishamco , I think that we don't have an easy way to test output files... using the real filesystem in the pipeline is not a good option. Do you have any suggestions?

**Comment by hishamco** [comment]
Let me try ..

**Comment by hishamco** [comment]
https://github.com/OrchardCMS/OrchardCore/blob/ba5e8c03c8f3aef138651184317e4859dab7adfd/src/OrchardCore/OrchardCore.Localization.Core/DefaultPluralRuleProvider.cs#L61-L73

**Comment by VincenzoCarlino** [comment]
Ok @hishamco I'll wait for your commit, let me know if I can do something.

**Comment by hishamco** [comment]
Sorry for the late, I will struggle to have a look today because we need to finalize this and publish a release before Orchard Core 2.0.0

**Comment by VincenzoCarlino** [comment]
> Sorry for the late, I will struggle to have a look today because we need to finalize this and publish a release before Orchard Core 2.0.0

Hello, any update?
Thanks @hishamco

**Comment by hishamco** [comment]
Seems I forgot this one, let me check it tonight if it's possible. My plan to publish a release ASAP and will be a part of OC 2.0.0

**Comment by hishamco** [comment]
I suggest trying this PR from `Program.cs` then we can focus on make the CLI commands testable

**Comment by hishamco** [comment]
@VincenzoCarlino please remove the changes in the test project, then I will merge, this not take so long to merge

**Review by hishamco — APPROVED** [review]

**Comment by hishamco** [comment]
Thanks for your contribution @VincenzoCarlino
